synapse/http/proxyagent.py

```diff
@@ -142,20 +150,35 @@ def __init__(
 
         self._federation_proxy_endpoint: Optional[IStreamClientEndpoint] = None
         if federation_proxies:
-            endpoints = []
+            endpoints: List[IStreamClientEndpoint] = []
             for federation_proxy in federation_proxies:
-                endpoint = HostnameEndpoint(
-                    self.proxy_reactor,
-                    federation_proxy.host,
-                    federation_proxy.port,
-                )
-
-                if federation_proxy.tls:
-                    tls_connection_creator = self._policy_for_https.creatorForNetloc(
+                endpoint: IStreamClientEndpoint
+                if isinstance(federation_proxy, InstanceTcpLocationConfig):
+                    endpoint = HostnameEndpoint(
+                        self.proxy_reactor,
                         federation_proxy.host,
                         federation_proxy.port,
                     )
-                    endpoint = wrapClientTLS(tls_connection_creator, endpoint)
+
+                    if federation_proxy.tls:
+                        tls_connection_creator = (
+                            self._policy_for_https.creatorForNetloc(
+                                federation_proxy.host,
+                                federation_proxy.port,
+                            )
+                        )
+                        endpoint = wrapClientTLS(tls_connection_creator, endpoint)
+
+                elif isinstance(federation_proxy, InstanceUnixLocationConfig):
+                    endpoint = UNIXClientEndpoint(
+                        self.proxy_reactor, federation_proxy.path
+                    )
+
+                else:
+                    # It is supremely unlikely we ever hit this
+                    raise SchemeNotSupported(
+                        f"Unknown type of Endpoint requested, check {federation_proxy}"
+                    )
```

Comment:
Some conflicts in this area because we had to revert the outbound federation proxy PR -> https://github.com/matrix-org/synapse/pull/15910

I can re-introduce these changes once we come up with a fix

Reply:
Used the GitHub UI to merge in `develop`. :crossed_fingers: hoping I did that right